### PR TITLE
Created models/tables, migrations

### LINKED ISSUE #9: Create Tables for DB
https://dbdiagram.io/d/5f4b4d2d88d052352cb54bc0

### PR DESCRIPTION
**Pull request by gabichuelas**
- Created `database/models.py` where I defined `Applicants`, `Messages`, `Values`, `Skills`, relevant join tables & relationships. 
- Ran `$ flask db init` to generate the `migrations` folder
- Anytime you run `$ python manage.py db migrate`, the `flask_migrate` package will generate a migration file, that you can then 'run' with `$ python manage.py db upgrade`. These will run on whatever database you specified last with `export DATABASE_URL...` from the database setup directions in the readme. 
- Currently have only 1 seed for an applicant. It lives in the `manage.py` file. It works because it shows up in the development db. 
- Added instructions to README concerning migrations 

- have done nothing with testing, not sure if that comes in yet. 

- Would like to figure out how to write a bunch of seeds and contain them in their own file, and seed the entire db structure with some examples. 

Closes #9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,9 +9,17 @@
 **Database Setup**
```
Comment (rrabinovitch) [inline README.md:12]: there should be instructions here about setting up the venv - I'll add this

**Comment by rrabinovitch** [comment]
I ran into the following error when trying any of the migration, upgrading, or seed commands: `ModuleNotFoundError: No module named 'psycopg2'`. After trying a few different approaches recommended [here](https://stackoverflow.com/questions/49811955/unable-to-install-psycopg2-pip-install-psycopg2), the solution that ended up working was `pip install psycopg2-binary`

File: api/database/models.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    __tablename__ = 'applicant_skills'
+    id = db.Column(db.Integer, primary_key=True)
+    applicant_id = db.Column(db.Integer, db.ForeignKey('applicants.id'))
+    skill_id = db.Column(db.Integer, db.ForeignKey('skills.id'))
+
+    created_at = Column(DateTime(timezone=True), nullable=False, default=datetime.now())
+
+    # relationships
+    applicant = db.relationship('Applicant', backref=db.backref("applicant_skills", cascade="all, delete-orphan"))
+    skill = db.relationship('Skill', backref=db.backref("applicant_skills", cascade="all, delete-orphan"))
```
Comment (rrabinovitch) [inline api/database/models.py:38]: can you share the resources you used to guide setting up these model relationships?

File: api/database/models.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+class Applicant(db.Model):
+    __tablename__ = 'applicants'
+
+    # The name of the column is the name you assign it to
+    id = Column(Integer, primary_key=True)
+    email = Column(String(100), unique=True, nullable=False)
+    first_name = Column(db.String(80))
+    last_name = Column(String(80))
+
```
Comment (rrabinovitch) [inline api/database/models.py:60]: What did we decide with @HireUp-Turing/front-end regarding how an applicant's name appears, if at all? Is that a column we need to include in the Applicants table?

File: api/database/models.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+class Message(db.Model):
+    __tablename__ = 'messages'
+
+    id = Column(Integer, primary_key=True)
+    employer_name = Column(String(100), nullable=False)
+    employer_email = Column(String(100), nullable=False)
+    body = Column(Text)
+
+    # Using Boolean:
+    read_status = Column(Boolean, default=False)
```
Comment (rrabinovitch) [inline api/database/models.py:80]: we should make sure @HireUp-Turing/front-end knows that at least for now we are using booleans for this attribute

File: pr_template.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#### What's this PR do?
```
Comment (rrabinovitch) [inline pr_template.md]: whoops, I'm sorry - I didn't realize you had included this in your PR, just that your PR wasn't using the template so I created one on `main`
I'm going to delete this one so it doesn't create a duplicate

**Review by rrabinovitch — APPROVED** [review]
thanks for getting this set up.
it would be helpful to include in PRs any resources we each use when implementing something new so that if the other one of us is needing to implement something similar, we're not spending as much time searching for guides. will you add notes on the resources you used for creating the migration file (I know you included the migration command in the PR, but I'm curious about the content of the file) and setting up the models?

File: README.md (hunk 1 above)
Comment (gabichuelas) [reply README.md:12]: There are. They're at the beginning of the README with a link to instructions on how to install python3 and pip3 with pyenv and setting those as the global versions so you don't have to type `python3` and `pip3` on every commands, or active or deactivate a virtual environment.

**Comment by gabichuelas** [comment]
> I ran into the following error when trying any of the migration, upgrading, or seed commands: `ModuleNotFoundError: No module named 'psycopg2'`. After trying a few different approaches recommended [here](https://stackoverflow.com/questions/49811955/unable-to-install-psycopg2-pip-install-psycopg2), the solution that ended up working was `pip install psycopg2-binary`

What you ended up doing with `pip install psycopg2-binary` was installing `psycopg2-binary` onto your local machine to make your commands work. (aka, it's not in the requirements.txt file, which is like the Gemfile in Flask), and therefore won't be available to anyone that clones down the repo.. BUT that's ok bc you don't need that package to run these commands. 

I suspect it has something to do with the differences between you running `venv` and me running `pyenv` - let's hop on a call soon and maybe get on the same page moving forward?

File: pr_template.md (hunk 5 above)
Comment (gabichuelas) [reply pr_template.md]: yeah I'd forgotten about the template, and added the template file to the repo after I'd already written my PR description.

File: api/database/models.py (hunk 4 above)
Comment (gabichuelas) [reply api/database/models.py:80]: Agreed - I was just trying to get something up that would work for this first go around

File: api/database/models.py (hunk 3 above)
Comment (gabichuelas) [reply api/database/models.py:60]: we didn't really decide. I wrote these columns based on the dbschema that you had created (the image I put in the readme for reference). We can always change this to suit later.

File: api/database/models.py (hunk 2 above)
Comment (gabichuelas) [reply api/database/models.py:38]: Yes, I've added a link to the docs for flask-sqlalchemy to the Notes: BE resources card

**Pull request merged**